instaloader/structures.py

```diff
@@ -275,6 +273,17 @@ def mediacount(self) -> int:
             return len(edges)
         return 1
 
+    def get_timestamp_date_created(self) -> float:
+        """Timestamp when the post was created"""
+        return (self._node["date"]
+                if "date" in self._node
+                else self._node["taken_at_timestamp"])
+
+    def get_timezone_offset(self, timestamp_date) -> float:
+        """Timestamp offset for a given date"""
+        diff_total_seconds = (datetime.fromtimestamp(timestamp_date) - datetime.utcfromtimestamp(timestamp_date)).total_seconds()
+        return diff_total_seconds / 3600
```

Comment:
Substracting  `datetime.utcfromtimestamp()` from `datetime.fromtimestamp()` produces a `datetime.timedelta` object which is needed inside the calculation of the `date_local` property to determine the `tzinfo`. Converting it forth and back into hours is not necessary and `get_timezone_offset()` could return the `timedelta` directly.

Since `get_timezone_offset()` does not use `self` and only uses the`timestamp_date` parameter to calculate the offset, it does not need to be a class method.

Reply:
I have just applied feedback as per the comments above, thank you!